x/programs/rust/examples/token/src/lib.rs

```diff
@@ -154,27 +154,24 @@ pub fn get_balance(context: Context<StateKeys>, recipient: Address) -> i64 {
 
 #[cfg(test)]
 mod tests {
-    use simulator::{Endpoint, Key, Param, Plan, Require, ResultAssertion, Step};
-
     use crate::INITIAL_SUPPLY;
+    use simulator::{Endpoint, Key, Param, Plan, Step};
 
     const PROGRAM_PATH: &str = env!("PROGRAM_PATH");
 
     #[test]
     fn create_program() {
-        let simulator = simulator::Client::new();
+        let mut simulator = simulator::Client::new_stdin();
```

Comment:
FYI, this comment isn't relevant, given my other comments, but in general, you shouldn't leak an abstraction in a function name like this. You're creating the client here; you shouldn't need to change the name if you change the internals (to use grpc instead of stdin for instance).

Reply:
I wanted to avoid the implementor to have to specify generics upfront, since we should only have a handful of transport layers, we probably don't need to be to allow that much customization. For instance, it probably doesn't make sense to have the writer being stdin and the responses being sent over IPC.

Reply:
This is the only communication layer available for now, we will abstract more once we write another one, see https://github.com/ava-labs/hypersdk/issues/939